### PR TITLE
feat(meta): adds InNamespace and WithAnnotation setter funcs

### PR DESCRIPTION
**Pull request by bartoszmajsak**
## Description

Adds new setter functions for manipulating MetaObject.

> [!NOTE]
> It is stacked PR coming from the ongoing authz work.

## Merge criteria

- [x] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [x] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [ ] ~Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.~
- [ ] ~Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).~
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: pkg/cluster/meta.go [hunk 1]
```diff
@@ -48,6 +48,26 @@ func WithLabels(labels ...string) MetaOptions {
...
+		obj.SetNamespace(ns)
+		return nil
+	}
+}
+
+func WithAnnotations(annotationKeyValue ...string) MetaOptions {
+	return func(obj metav1.Object) error {
+		annotationsMap, err := extractKeyValues(annotationKeyValue)
+		if err != nil {
+			return fmt.Errorf("failed unable to set labels: %w", err)
```
Comment (zdtsw) [inline pkg/cluster/meta.go:62]:
```suggestion
			return fmt.Errorf("failed to set labels: %w", err)
```

File: pkg/cluster/meta.go [hunk 2]
```diff
@@ -48,6 +48,26 @@ func WithLabels(labels ...string) MetaOptions {
 	}
 }
 
+func InNamespace(ns string) MetaOptions {
```
Comment (zdtsw) [inline pkg/cluster/meta.go:51]: these 2 are placeholders not get called yet, right?

File: pkg/cluster/resources.go [hunk 3]
```diff
@@ -88,6 +88,10 @@ func CreateSecret(ctx context.Context, cli client.Client, name, namespace string
 // If the configmap already exists, it will be updated with the merged Data and MetaOptions, if any.
 // ConfigMap.ObjectMeta.Name and ConfigMap.ObjectMeta.Namespace are both required, it returns an error otherwise.
 func CreateOrUpdateConfigMap(ctx context.Context, c client.Client, desiredCfgMap *corev1.ConfigMap, metaOptions ...MetaOptions) error {
+	if applyErr := ApplyMetaOptions(desiredCfgMap, metaOptions...); applyErr != nil {
```
Comment (zdtsw) [inline pkg/cluster/resources.go:91]: should this be called after "`if desiredCfgMap.GetName() == "" || desiredCfgMap.GetNamespace() == "" {`" ?

File: pkg/cluster/meta.go (hunk 2 above)
Comment (bartoszmajsak) [reply pkg/cluster/meta.go:51]: Yeah, as mentioned in the description. Trying to make the final PR as small as possible.

File: pkg/cluster/resources.go (hunk 3 above)
Comment (bartoszmajsak) [reply pkg/cluster/resources.go:91]: I think if we change the order you would never be able to pass MetaOptions which could set those fields (i.e. InNamespace)

Comment (zdtsw) [reply pkg/cluster/resources.go:91]: right, `GetNamespace` will be different with the order or calls.

**Review by zdtsw — APPROVED** [review]

**Comment by zdtsw** [comment]
feel to add a "onhold" if need others review

**Comment by bartoszmajsak** [comment]
@zdtsw I think I'm good, thanks for taking care of review.

**Pull request merged**